src/RestWrite.js

```diff
@@ -1482,20 +1482,25 @@ RestWrite.prototype.objectId = function() {
 };
 
 // Returns a copy of the data and delete bad keys (_auth_data, _hashed_password...)
-RestWrite.prototype.sanitizedData = function() {
+RestWrite.prototype.sanitizedData = function(decodeData = true) {
   const data = Object.keys(this.data).reduce((data, key) => {
     // Regexp comes from Parse.Object.prototype.validate
     if (!/^[A-Za-z][0-9A-Za-z_]*$/.test(key)) {
       delete data[key];
     }
     return data;
   }, deepcopy(this.data));
-  return Parse._decode(undefined, data);
+  if (decodeData) {
+    // decoded data might contain instances of ParseObject, ParseRelation, ParseACl...
+    return Parse._decode(undefined, data);
+  }
+  // data is kept in json format. Not decoded
+  return data;
 };
 
 // Returns an updated copy of the object
 RestWrite.prototype.buildUpdatedObject = function(extraData) {
-  const updatedObject = triggers.inflate(extraData, this.originalData);
+  let updatedObject = triggers.inflate(extraData, this.originalData);
   Object.keys(this.data).reduce(function(data, key) {
```

Comment:
I did not really understand what is the purpose of this `reduce`.
Didn't touch it, but `RestWrite.prototype.buildUpdatedObject = function(extraData) {` _could_ be fixed to have less duplicated code (like inflating twice in the case of session)

Reply:
Please revert to const